pkg/component/feature/feature.go

```diff
@@ -116,11 +153,17 @@ func NewSecurityFeature(opts *Options) *SecurityFeature {
 	cff := &CommonFeatureFields{
 		Options: *opts,
 	}
-	cff.components = []component.IstioComponent{
-		component.NewCitadelComponent(newComponentOptions(cff, name.SecurityFeatureName)),
-		component.NewCertManagerComponent(newComponentOptions(cff, name.SecurityFeatureName)),
-		component.NewNodeAgentComponent(newComponentOptions(cff, name.SecurityFeatureName)),
+
+	if opts != nil && opts.Traslator != nil && opts.Traslator.FeatureMaps != nil {
```

Comment:
This code looks to be repeated - could you put it in a function?

Reply:
Good idea, done.